### PR TITLE
openconnect-head: init at 2021-05-05

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change

Needed fortinet and IPv6 support which is in HEAD but not in 8.10 (the latest stable release)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change
- [x] Tested execution of all binary files
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122617` at 2d1090a0 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>openconnect_head</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`lgpl21` is a deprecated license, please check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/tools/networking/openconnect/default.nix:58:5:
```
   |
58 |     license = licenses.lgpl21;
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/tools/networking/openconnect/default.nix [hunk 1]
```diff
@@ -1,4 +1,19 @@
...
+, gnutls ? null
+, gmp
+, libxml2
+, stoken
+, zlib
+, fetchgit
+, darwin
+, head ? false
+  , fetchFromGitLab
+  , autoreconfHook
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/openconnect/default.nix:15]:
```suggestion
, head ? false
, fetchFromGitLab
, autoreconfHook
```

File: pkgs/tools/networking/openconnect/default.nix [hunk 2]
```diff
@@ -9,15 +24,20 @@ let vpnc = fetchgit {
...
-    sha256 = "1cdsx4nsrwawbsisfkldfc9i4qn60g03vxb13nzppr2br9p4rrih";
-  };
+  pname = "openconnect${lib.optionalString head "-head"}";
+  version = if head then "2021-05-05" else "8.10";
+
+  src =
+    if head then fetchFromGitLab {
+      owner = "openconnect";
+      repo = "openconnect";
+      rev = "684f6db1aef78e61e01f511c728bf658c30b9114";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/openconnect/default.nix:34]: That rev does not exist https://github.com/openconnect/openconnect/commit/684f6db1aef78e61e01f511c728bf658c30b9114

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -9278,6 +9278,11 @@ in
     openssl = null;
   };
 
+  openconnect_head = callPackage ../tools/networking/openconnect {
+    head = true;
+    openssl = null;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:9283]:
```suggestion
```

Comment (rnhmjoj) [reply pkgs/top-level/all-packages.nix:9283]: Doing so would probably break the build, see the lines above.

File: pkgs/tools/networking/openconnect/default.nix (hunk 2 above)
Comment (rnhmjoj) [reply pkgs/tools/networking/openconnect/default.nix:34]: openconnect is hosted on gitlab.com, not github.

https://gitlab.com/openconnect/openconnect/-/commit/684f6db1aef78e61e01f511c728bf658c30b9114

File: pkgs/tools/networking/openconnect/default.nix (hunk 1 above)
Comment (rnhmjoj) [reply pkgs/tools/networking/openconnect/default.nix:15]: Again, I use an indent as a cue that these dependencies as conditioned on `head == true`.

Comment (SuperSandro2000) [reply pkgs/tools/networking/openconnect/default.nix:15]: That will be removed when formatting with either nixpkgs-fmt or nixfmt.

**Pull request merged**